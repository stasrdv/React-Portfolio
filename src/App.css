.col-sm-10 {
  min-height: 650px;
}
