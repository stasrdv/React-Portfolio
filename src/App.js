import React, { Component } from "react";
import { VerticalNavbar } from "./components/vertical-navbar";
import { PorfolioBoard } from "./components/portfoliob-board";
import { AboutPage } from "./components/about";
import "./App.css";

class App extends React.Component {
  ActiveComponent = PorfolioBoard;
  constructor(props) {
    super(props);
    this.state = {
      isAboutPageActive: false
    };
  }

  toggleDisplay = event => {
    this.setState({ isAboutPageActive: !this.state.isAboutPageActive });
  };

  render() {
    if (!this.state.isAboutPageActive) {
      this.ActiveComponent = PorfolioBoard;
    } else {
      this.ActiveComponent = AboutPage;
    }
    return (
      <div className="container-fluid">
        <div className="row">
          <VerticalNavbar toggleDisplay={this.toggleDisplay} />
          <this.ActiveComponent />
        </div>
      </div>
    );
  }
}

export default App;
