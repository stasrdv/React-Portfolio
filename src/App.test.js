import React from "react";
import App from "./App";
import { shallow } from "enzyme";
import { storiesOf } from "@storybook/react";

describe("App Component", () => {
  it("renders without crashing", () => {
    shallow(<App />);
  });

  it('should render correctly in "debug" mode', () => {
    const component = shallow(<App debug />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with no props", () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });

  it("should render", () => {
    const component = shallow(<App />);
    expect(component).toHaveLength(1);
  });
});
