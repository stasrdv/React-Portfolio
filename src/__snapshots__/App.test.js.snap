// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render correctly in "debug" mode 1`] = `
<BrowserRouter>
  <Route
    render={[Function]}
  />
</BrowserRouter>
`;

exports[`App Component should render correctly with no props 1`] = `
<BrowserRouter>
  <Route
    render={[Function]}
  />
</BrowserRouter>
`;

exports[`App Componentcl should render correctly in "debug" mode 1`] = `
<BrowserRouter>
  <Route
    render={[Function]}
  />
</BrowserRouter>
`;

exports[`App Componentcl should render correctly with no props 1`] = `
<BrowserRouter>
  <Route
    render={[Function]}
  />
</BrowserRouter>
`;

exports[`My App should render   correctly with given items 1`] = `
<div
  className="jumobtron board"
>
  <div
    className="row justify-content-center"
  >
    <Card
      deleteItem={[Function]}
      description="React native todolist implementation"
      id={4}
      img="https://raw.githubusercontent.com/lucasbento/react-native-actions/master/common/media/logo.png"
      isCartMode={true}
      key="4"
      title="React-Native-ToDoList"
      url="https://github.com/stasrdv/React-Native-ToDoList"
    />
    <Card
      deleteItem={[Function]}
      description="WebSockets simple chat example (Socket.io ,ExpressJs,VanillaJS)"
      id={1}
      img="https://cdn-images-1.medium.com/max/800/0*xAADmPJN52Yy6XJV.jpg"
      isCartMode={true}
      key="1"
      title="WebSockets-MultiChat"
      url="https://github.com/stasrdv/WebSockets-MultiChat-Example"
    />
  </div>
</div>
`;

exports[`My App should render correctly in "debug" mode 1`] = `
<BrowserRouter>
  <Route
    render={[Function]}
  />
</BrowserRouter>
`;

exports[`My App should render correctly with no props 1`] = `
<BrowserRouter>
  <Route
    render={[Function]}
  />
</BrowserRouter>
`;
